java/vector/src/test/java/org/apache/arrow/vector/ipc/TestJSONFile.java

```diff
@@ -300,30 +296,25 @@ public void testWriteReadDecimalJSON() throws IOException {
     File file = new File("target/mytest_decimal.json");
 
     // write
-    try (
-        BufferAllocator vectorAllocator = allocator.newChildAllocator("original vectors", 0, Integer.MAX_VALUE)
-    ) {
-
-      try (VectorSchemaRoot root = writeDecimalData(vectorAllocator)) {
-        printVectors(root.getFieldVectors());
-        validateDecimalData(root);
-        writeJSON(file, root, null);
-      }
+    try (BufferAllocator vectorAllocator = allocator.newChildAllocator("original vectors", 0, Integer.MAX_VALUE);
+        VectorSchemaRoot root = writeDecimalData(vectorAllocator)) {
+      printVectors(root.getFieldVectors());
+      validateDecimalData(root);
+      writeJSON(file, root, null);
     }
 
     // read
     try (
         BufferAllocator readerAllocator = allocator.newChildAllocator("reader", 0, Integer.MAX_VALUE);
+        JsonFileReader reader = new JsonFileReader(file, readerAllocator)
     ) {
-      JsonFileReader reader = new JsonFileReader(file, readerAllocator);
       Schema schema = reader.start();
       LOGGER.debug("reading schema: " + schema);
 
       // initialize vectors
       try (VectorSchemaRoot root = reader.read();) {
         validateDecimalData(root);
       }
-      reader.close();
```

Comment:
does this really fail on style or is this just separate cleanup?

Reply:
The reader is now closed by the try-with-resources pattern: https://github.com/apache/arrow/pull/6186/files#diff-1ed18d6881a6b20dd9f67c9878605613R309

A lot of the readers in this file weren't being closed, so I've moved them to the try-with-resources block.